cmake/module/Maintenance.cmake

```diff
@@ -30,11 +30,15 @@ function(add_maintenance_targets)
   else()
     set(exe_format ELF)
   endif()
-  if(CMAKE_CROSSCOMPILING)
-    list(JOIN DEPENDS_C_COMPILER_WITH_LAUNCHER " " c_compiler_command)
-  else()
-    set(c_compiler_command ${CMAKE_C_COMPILER})
-  endif()
+
+  # In CMake, the components of the compiler invocation are separated into three distinct variables:
+  #  - CMAKE_C_COMPILER_LAUNCHER: a semicolon-separated list of launchers or tools to precede the compiler (e.g., env or ccache).
```

Comment:
Does this mean that ccache and env can't be used simultaneously? Basically meaning that ccache can't be used for darwin builds?

Reply:
> Does this mean that ccache and env can't be used simultaneously?

No, it does not.

The current implementation works flawlessly for ccache + darwin. It appends ccache command to the `CMAKE_{C.CXX}_COMPILER_LAUNCHER` variables after `env`.